Dockerfile

```
@@ -16,7 +16,7 @@ ENV PATH=/venv/bin:$PATH
 FROM developer as build
 COPY . /context
 WORKDIR /context
-RUN pip install .
+RUN touch dev-requirements.txt && pip install --upgrade pip && pip install -r dev-requirements.txt
```

As I said in https://github.com/DiamondLightSource/blueapi/pull/499#issuecomment-2158961600

please make this:
```suggestion
RUN touch dev-requirements.txt && pip install --upgrade pip && pip install -c dev-requirements.txt .
```
```